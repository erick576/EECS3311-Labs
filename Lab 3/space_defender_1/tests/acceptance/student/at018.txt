play(5, 10, 5, 1)
move(A, 1)
move(E, 1)
move(B, 1)
move(C, 1)
move(C, 3)
move(C, 2)
move(C, 1)
move(A, 2)
move(C, 1)
move(E, 2)
move(C, 1)
move(C, 3)
move(E, 1)
move(C, 3)
move(A, 1)
move(C, 3)
fire
move(C, 6)
pass
play(10, 10 , 10, 5)
fire
pass
pass
pass
fire
move(J, 3)
fire
move(A, 3)
fire
fire
pass
pass
move(B, 2)
fire
move(B, 8)
