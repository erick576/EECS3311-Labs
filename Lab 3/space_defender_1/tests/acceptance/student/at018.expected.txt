  Welcome to Space Defender Version 1.
->play(5,10,5,1)
  state:1.0, ok
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:2.0, ok
  The Starfighter moves: [C,1] -> [A,1]
      1  2  3  4  5  6  7  8  9 10
    A S  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(E,1)
  state:3.0, ok
  The Starfighter moves: [A,1] -> [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
->move(B,1)
  state:4.0, ok
  The Starfighter moves: [E,1] -> [B,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B S  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,1)
  state:5.0, ok
  The Starfighter moves: [B,1] -> [C,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,3)
  state:6.0, ok
  The Starfighter moves: [C,1] -> [C,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,2)
  state:7.0, ok
  The Starfighter moves: [C,3] -> [C,2]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  S  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,1)
  state:8.0, ok
  The Starfighter moves: [C,2] -> [C,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,2)
  state:9.0, ok
  The Starfighter moves: [C,1] -> [A,2]
      1  2  3  4  5  6  7  8  9 10
    A _  S  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,1)
  state:10.0, ok
  The Starfighter moves: [A,2] -> [C,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(E,2)
  state:11.0, ok
  The Starfighter moves: [C,1] -> [E,2]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  S  _  _  _  _  _  _  _  _
->move(C,1)
  state:12.0, ok
  The Starfighter moves: [E,2] -> [C,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,3)
  state:13.0, ok
  The Starfighter moves: [C,1] -> [C,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(E,1)
  state:14.0, ok
  The Starfighter moves: [C,3] -> [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
->move(C,3)
  state:15.0, ok
  The Starfighter moves: [E,1] -> [C,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:16.0, ok
  The Starfighter moves: [C,3] -> [A,1]
      1  2  3  4  5  6  7  8  9 10
    A S  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,3)
  state:17.0, ok
  The Starfighter moves: [A,1] -> [C,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:18.0, ok
  The Starfighter fires a projectile at: [C,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  S  *  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(C,6)
  state:19.0, ok
  A projectile moves: [C,4] -> [C,5]
  The Starfighter moves and collides with a projectile: [C,3] -> [C,5]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  X  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
->pass
  state:19.1, error
  Not in game.
->play(10,10,10,5)
  state:20.0, ok
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->fire
  state:21.0, ok
  The Starfighter fires a projectile at: [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  *  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:22.0, ok
  A projectile moves: [E,2] -> [E,7]
  The Starfighter stays at: [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  *  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:23.0, ok
  A projectile moves: [E,7] -> out of the board
  The Starfighter stays at: [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:24.0, ok
  The Starfighter stays at: [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->fire
  state:25.0, ok
  The Starfighter fires a projectile at: [E,1]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  *  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(J,3)
  state:26.0, ok
  A projectile moves: [E,2] -> [E,7]
  The Starfighter moves: [E,1] -> [J,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  *  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  S  _  _  _  _  _  _  _
->fire
  state:27.0, ok
  A projectile moves: [E,7] -> out of the board
  The Starfighter fires a projectile at: [J,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  S  *  _  _  _  _  _  _
->move(A,3)
  state:28.0, ok
  A projectile moves: [J,4] -> [J,9]
  The Starfighter moves: [J,3] -> [A,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  S  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  *  _
->fire
  state:29.0, ok
  A projectile moves: [J,9] -> out of the board
  The Starfighter fires a projectile at: [A,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  S  *  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->fire
  state:30.0, ok
  A projectile moves: [A,4] -> [A,9]
  The Starfighter fires a projectile at: [A,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  S  *  _  _  _  _  *  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:31.0, ok
  A projectile moves: [A,9] -> out of the board
  A projectile moves: [A,4] -> [A,9]
  The Starfighter stays at: [A,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  S  _  _  _  _  _  *  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:32.0, ok
  A projectile moves: [A,9] -> out of the board
  The Starfighter stays at: [A,3]
      1  2  3  4  5  6  7  8  9 10
    A _  _  S  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(B,2)
  state:33.0, ok
  The Starfighter moves: [A,3] -> [B,2]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  S  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->fire
  state:34.0, ok
  The Starfighter fires a projectile at: [B,2]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  S  *  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(B,8)
  state:35.0, ok
  A projectile moves: [B,3] -> [B,8]
  The Starfighter moves and collides with a projectile: [B,2] -> [B,8]
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  X  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
